packages/ag-charts-community/src/module/optionsModule.ts

```diff
@@ -157,12 +157,18 @@ export class ChartOptions<T extends AgChartOptions = AgChartOptions> {
             this.removeLeftoverSymbols(this.userOptions);
         }
 
+        const dataChangedLength =
+            currentUserOptions instanceof ChartOptions &&
+            deltaOptions?.data !== undefined &&
+            deltaOptions?.data?.length !== currentUserOptions.userOptions.data?.length;
+
         let activeTheme, processedOptions, fastDelta, themeParameters, annotationThemes, googleFonts, optionsGraph;
         if (
             !stripSymbols &&
             deltaOptions !== undefined &&
             ChartOptions.isFastPathDelta(deltaOptions) &&
-            baseChartOptions != null
+            baseChartOptions != null &&
+            !dataChangedLength
```

Comment:
This ensures the slow setup is taken if the data has changed length. This is required to allow the `$applyCycle` operation to update with the new data length. Otherwise if datums are added/removed it would not re-evaluate the options graph.

Is this ok from an optimisation point-of-view?

Reply:
SGTM